src/main/java/io/github/syst3ms/skriptparser/registration/SkriptRegistration.java

```diff
@@ -662,7 +677,11 @@ protected List<PatternElement> parsePatterns() {
             boolean computePriority = priority == -1;
             priority = computePriority ? 5 : priority;
             return patterns.stream()
-                    .map(s -> PatternParser.parsePattern(s, logger).orElse(null))
+                    .map(s -> {
+                        var result = PatternParser.parsePattern(s, logger).orElse(null);
+                        logger.finalizeLogs();
```

Comment:
Shouldn't this logger be outside the stream? After all the patterns in the map have parsed, then finalize the logs?

Reply:
If I recall correctly, finalizing the logs should be done after each parsing attempt, so if it is a pattern or a line of code. The naming could be a bit confusing, but honestly the whole SkriptLogger class lacks propper documentation and it took me a while to figure out as well. 

I'll take a look just to be sure.